lib/models/asset_token.dart

```diff
@@ -5,178 +5,370 @@
-        sourceURL: asset.projectMetadata.latest.sourceUrl,
-        thumbnailID: asset.thumbnailID,
-        thumbnailURL: asset.projectMetadata.latest.thumbnailUrl,
-        galleryThumbnailURL: asset.projectMetadata.latest.galleryThumbnailUrl,
-        title: asset.projectMetadata.latest.title,
-        initialSaleModel: asset.saleModel,
-        ownerAddress: asset.owner,
-        owners: asset.owners,
-        balance: asset.balance,
-        lastActivityTime: asset.lastActivityTime,
-        provenances: asset.provenance,
-        isFeralfileFrame:
-            asset.projectMetadata.latest.artworkMetadata?["isFeralfileFrame"],
-        pending: false,
-        originTokenInfoId: asset.originTokenInfo?.firstOrNull?.id,
-        swapped: asset.swapped ?? false,
-        scrollable: asset.attributes?.scrollable,
-      );
+  String id;
 
-  @override
-  bool operator ==(Object other) =>
-      identical(this, other) ||
-      other is AssetToken &&
-          runtimeType == other.runtimeType &&
-          id == other.id &&
-          pending == other.pending;
+  int? balance;
+  String owner;
 
-  @override
-  int get hashCode => id.hashCode;
+  DateTime lastActivityTime;
+  DateTime lastRefreshedTime;
 
-  DateTime get lastUpdateTime => updateTime ?? lastActivityTime;
-}
+  bool? pending;
+  bool? isDebugged;
 
-class TokenOwnersConverter extends TypeConverter<Map<String, int>, String> {
-  @override
-  Map<String, int> decode(String? databaseValue) {
-    if (databaseValue?.isNotEmpty == true) {
-      return (json.decode(databaseValue!) as Map<String, dynamic>?)
-              ?.map((key, value) => MapEntry(key, (value as int?) ?? 0)) ??
-          {};
-    } else {
-      return {};
-    }
+  String? mimeType;
+  String? previewURL;
+  String? thumbnailURL;
+  String? thumbnailID;
+  String? galleryThumbnailURL;
+  String? artistID;
+  String? blockchain;
+  String? tokenId;
+  String? title;
+
+  factory CompactedAssetToken.fromAssetToken(AssetToken assetToken) {
+    return CompactedAssetToken(
+      id: assetToken.id,
+      balance: assetToken.balance,
+      owner: assetToken.owner,
+      lastActivityTime: assetToken.lastActivityTime,
+      lastRefreshedTime: assetToken.lastRefreshedTime,
+      mimeType: assetToken.mimeType,
+      previewURL: assetToken.previewURL,
+      thumbnailURL: assetToken.thumbnailURL,
+      thumbnailID: assetToken.thumbnailID,
+      galleryThumbnailURL: assetToken.galleryThumbnailURL,
+      pending: assetToken.pending,
+      isDebugged: assetToken.isDebugged,
+      artistID: assetToken.artistID,
+      blockchain: assetToken.blockchain,
+      tokenId: assetToken.tokenId,
+      title: assetToken.title,
+    );
   }
 
   @override
-  String encode(Map<String, int>? value) {
-    if (value == null) {
-      return "{}";
-    } else {
-      return json.encode(value);
-    }
+  int compareTo(other) {
+    return other.lastActivityTime.compareTo(lastActivityTime);
```

Comment:
I think you might need to add the `id` into this compare function for the case `lastActivityTime`s are identical.